### PR TITLE
Add new Python API post

### PR DESCRIPTION
**Pull request by aturley**
This blog post describes our new Python API.

### CODE REVIEW

File: content/post/new-python-api.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+We took the feedback and began to think about what kinds of things made the existing API unpythonic. Then we mocked up some of our ideas and tried them out by reimplementing some of our example programs to see how they felt. Finally, we worked on solving some unexpected problems that arose from the changes. The end result is a new Wallaroo Python API that is much more concise and Pythonic than the original. We’ve gotten some great early feedback, and we think it represents a notable improvement over the original API, and we think that Python programmers will agree.
```
Comment (dipinhora) [inline content/post/new-python-api.md]: The two uses of `and we think` sounds a bit awkward. Maybe rephrase to:

> We’ve gotten some great early feedback indicating a notable improvement over the original API, and we think that Python programmers will agree.

File: content/post/new-python-api.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+Before we had even published the blog post about the original Python API, we had discussed ideas for things that we could do to improve the API. One of the ideas that we had considered was using decorators to cut down on the number of classes and thus reduce the amount of code that needed to be written. When we asked for feedback, several people suggested using decorators to improve the API, so we felt that our earlier idea had been validated. We went ahead and designed a new decorator-based API.
+
+## The New Python API
+
+### A Motivating Example
+
+We will start with the canonical streaming data processing application, Word Count. A stream of input text is analyzed and the total number of times each word has been seen is reported. [The example](https://github.com/WallarooLabs/wallaroo/blob/release/examples/python/word_count/word_count.py) in it's entirety is in our [GitHub repository](https://github.com/WallarooLabs/wallaroo/tree/release).
+
+We will make the following assumptions:
+* Incoming messages will come from a TCP connection and be sent to
```
Comment (dipinhora) [inline content/post/new-python-api.md]: These bullets didn't render correctly in the preview.

File: content/post/new-python-api.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+### Wallaroo's Core Abstractions
+
+In order to understand the Python API, it is important to understand Wallaroo's core abstractions:
+
+* State -- Accumulated result of data stored over the course of time.
+* Computation -- Code that transforms an input to an
+  output.
+* State Computation -- Code that takes an input and a state
+  object, operates on that input and state
+  (possibly making state updates), and optionally produces an output.
```
Comment (dipinhora) [inline content/post/new-python-api.md]: We use `state` and `state object` here and then later we also use `state partition` and `state entity`. It seems there is at least some ambiguity between `state object` and `state entity`. Maybe they're supposed to be the same thing?

File: content/post/new-python-api.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+#### State Partitions and State Entities
+
+In this example, the state is the number of times each word has been seen. The easiest way to do this would be with a dictionary where the key is a word, and the value associated with that key is the number of times that word has been seen in the event stream.
+
+Wallaroo lets you divide state partitions into pieces called state entities. State entities are pieces of state partitions that are uniquely identified by a key of some sort. State can be partitioned by any number of keys. The only restriction is that the state entities must be completely isolated from each other so that they can be accessed and updated independently.
+
+When a message is sent, Wallaroo applies a partition function to the message to determine which state partition to send it to. Different state entities may live on different workers, and an entity may move from one worker to another when workers are added or removed from the cluster. This makes it easy to scale the application up and down as the number of workers in the cluster increases and decreases.
```
Comment (dipinhora) [inline content/post/new-python-api.md:134]: There seems to be a jump between `state partition` and `state entity` here. The first sentence says that the `partition function to the message to determine which state partition to send it to` but then the next sentence starts talking about `state entities` living on different workers. I know the previous paragraph defined that the `state entities` are subsets of the `state partitions` but this still feels off to me.

File: content/post/new-python-api.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+`count_word` is a state computation; it uses an incoming message and a state entity to update the word count for the new word and returns a message for Wallaroo to send on its behalf.
+
+```python
+@wallaroo.state_computation(name="Count Word")
+def count_word(word, word_totals):
+    word_totals.update(word)
+    return (word_totals.get_count(word), True)
+```
+
+The `@wallaroo.state_computation(...)` decorator must be used to indicate that this is a state computation. As with the computation above, the `name` argument specifies the name that Wallaroo will use when reporting information about the application.
```
Comment (dipinhora) [inline content/post/new-python-api.md:215]: It might be useful to mention why the `True` is returned.

File: content/post/new-python-api.md (hunk 1 above)
Comment (aturley) [reply content/post/new-python-api.md]: Removing second one.

File: content/post/new-python-api.md (hunk 2 above)
Comment (aturley) [reply content/post/new-python-api.md]: Fixing list.

File: content/post/new-python-api.md (hunk 3 above)
Comment (aturley) [reply content/post/new-python-api.md]: Changing "state object" to "state entity".

File: content/post/new-python-api.md (hunk 5 above)
Comment (aturley) [reply content/post/new-python-api.md:215]: I'll add a clarification.
